openscenario/openscenario_interpreter/include/openscenario_interpreter/utility/execution_timer.hpp

```diff
@@ -103,11 +48,28 @@ class ExecutionTimer
 
     const auto end = Clock::now();
 
-    statistics_map[tag].add(end - begin);
+    statistics_map[tag](std::chrono::duration_cast<std::chrono::nanoseconds>(end - begin).count());
 
     return end - begin;
   }
 
+  void saveStatistics(boost::filesystem::path output_file)
+  {
+    nlohmann::json json_data;
+    for (const auto & [name, statistics] : statistics_map) {
+      json_data[name + "/min"] = boost::accumulators::extract::min(statistics) * 1e-9;
+      json_data[name + "/max"] = boost::accumulators::extract::max(statistics) * 1e-9;
+      json_data[name + "/mean"] = boost::accumulators::extract::mean(statistics) * 1e-9;
+      json_data[name + "/stddev"] =
+        std::sqrt(boost::accumulators::extract::variance(statistics)) * 1e-9;
+      json_data[name + "/count"] = boost::accumulators::extract::count(statistics);
+    }
+
+    std::ofstream file(output_file);
+    file << json_data.dump(4);
+    file.close();
```

Comment:
This line is not necessary because the destructor of std::ofstream automatically calls close (if the file is open).

```suggestion
```

Reply:
fixed in [`6e9a448` (#1537)](https://github.com/tier4/scenario_simulator_v2/pull/1537/commits/6e9a448a5c428bc59d74c0daaa2774d2d796cf2f)